backoff.c

```diff
@@ -0,0 +1,87 @@
+#include "common.h"
+
+#include <ext/standard/php_rand.h>
+#include <ext/standard/php_mt_rand.h>
+
+#include "backoff.h"
+
+zend_long random_range(zend_long min, zend_long max) {
+    if (max <= min) {
+        max = min + 1;
+    }
```

Comment:
I think this might end up causing confusion in the future.

Perhaps we can just make sure we never call when max <= min?